Source/items.cpp

```diff
@@ -3106,44 +3106,31 @@ void CreatePlrItems(int playerId)
 
 bool ItemSpaceOk(Point position)
 {
-	if (!InDungeonBounds(position))
+	if (!InDungeonBounds(position)) {
 		return false;
+	}
 
-	if (dMonster[position.x][position.y] != 0)
+	if (IsTileSolid(position)) {
 		return false;
+	}
 
-	if (dPlayer[position.x][position.y] != 0)
+	if (dItem[position.x][position.y] != 0) {
 		return false;
+	}
 
-	if (dItem[position.x][position.y] != 0)
+	if (dMonster[position.x][position.y] != 0) {
 		return false;
-
-	if (dObject[position.x][position.y] != 0) {
-		int oi = abs(dObject[position.x][position.y]) - 1;
-		if (Objects[oi]._oSolidFlag)
-			return false;
 	}
 
-	Point south = position + Direction::South;
-	if (InDungeonBounds(south)) {
-		int objectId = dObject[south.x][south.y];
-		if (objectId != 0 && Objects[abs(objectId) - 1]._oSelFlag != 0)
-			return false;
+	if (dPlayer[position.x][position.y] != 0) {
+		return false;
 	}
 
-	Point southEast = position + Direction::SouthEast;
-	Point southWest = position + Direction::SouthWest;
-	if (InDungeonBounds(southEast) && InDungeonBounds(southWest)) {
-		int objectIdSE = dObject[southEast.x][southEast.y];
-		int objectIdSW = dObject[southWest.x][southWest.y];
-		if (objectIdSE > 0 && objectIdSW > 0) {
-			if (Objects[objectIdSE - 1]._oSelFlag != 0 && Objects[objectIdSW - 1]._oSelFlag != 0) {
-				return false;
-			}
-		}

```

Comment:
Is all of this unessesery because the dObject will contain the negative id at the base position anyway?

Reply:
Rolled into `IsItemBlockingObjectAtPosition`. This code was identical to the checks in inv.cpp `CanPut`. The `considerLargeObjects` flag is set to false for this check to replicate the positive index only behaviour.